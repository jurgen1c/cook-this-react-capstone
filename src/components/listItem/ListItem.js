import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { getSingleRecipe } from '../../actions';
import './listItem.css';

const ListItem = ({ meal, dispatch }) => {
  const handleChoice = () => {
    getSingleRecipe(meal.idMeal, dispatch);
  };

  return (
    <div className="list-item">
      <img src={meal.strMealThumb} alt="meal-thumb" style={{ width: '50px', height: '50px' }} />
      <div>
        <h2>{meal.strMeal}</h2>
        <Link to={`/${meal.strMeal.toLowerCase()}`} onClick={handleChoice}> Start Cooking</Link>
      </div>
    </div>
  );
};

ListItem.propTypes = {
  meal: PropTypes.exact(
    {
      idMeal: PropTypes.string,
      strMeal: PropTypes.string,
      strMealThumb: PropTypes.string,
    },
  ).isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default ListItem;
