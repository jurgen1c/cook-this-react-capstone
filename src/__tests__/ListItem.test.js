import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../components/listItem/ListItem';

test('Componenet renders correctly', () => {
  const component = renderer.create(
    <ListItem />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});