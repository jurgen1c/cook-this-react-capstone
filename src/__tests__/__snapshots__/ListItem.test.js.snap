// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componenet renders correctly 1`] = `
Array [
  <div
    className="list-item"
  >
    <img
      alt="meal-thumb"
      src="mealIamageUrl"
    />
    <div>
      <h2>
        Meal
      </h2>
      <a
        className="btn"
        href="/meal"
        onClick={[Function]}
      >
         Start Cooking
      </a>
    </div>
  </div>,
  ",",
]
`;
